File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/scene/SceneController.java
package com.valentingrigorean.arcgis_maps_flutter.scene;

import com.esri.arcgisruntime.mapping.ArcGISScene;
import com.esri.arcgisruntime.mapping.Surface;
import com.esri.arcgisruntime.mapping.view.SceneView;
import com.valentingrigorean.arcgis_maps_flutter.Convert;

public class SceneController {
    private SceneView sceneView;

    private ArcGISScene scene;

    private Surface surface;

    public void setSceneView(SceneView sceneView) {

        this.sceneView = sceneView;
    }

    public void setScene(Object json) {
        if (json == null) {
            return;
        }

        scene = Convert.toScene(json);
        if (surface != null) {
            scene.setBaseSurface(surface);
        }

        sceneView.setScene(scene);
    }

    public void setSurface(Object json) {
        if (json == null) {
            return;
        }
        surface = Convert.toSurface(json);
        if (scene != null) {
            scene.setBaseSurface(surface);
        }
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/scene/SceneController.kt
package com.valentingrigorean.arcgis_maps_flutter.scene

import com.arcgismaps.mapping.ArcGISScene
import com.arcgismaps.mapping.Surface
import com.arcgismaps.mapping.view.SceneView

import com.valentingrigorean.arcgis_maps_flutter.Convert

class SceneController {
    private var sceneView: SceneView? = null
    private var scene: ArcGISScene? = null
    private var surface: Surface? = null
    fun setSceneView(sceneView: SceneView?) {
        this.sceneView = sceneView
    }

//    fun setScene(json: Any?) {
//        if (json == null) {
//            return
//        }
//        scene = Convert.Companion.toScene(json)
//        if (surface != null) {
//            scene!!.baseSurface = surface
//        }
//        sceneView!!.scene = scene
//    }
//
//    fun setSurface(json: Any?) {
//        if (json == null) {
//            return
//        }
//        surface = Convert.Companion.toSurface(json)
//        if (scene != null) {
//            scene!!.baseSurface = surface
//        }
//    }
}